modules/caas/api/src/main/java/io/cattle/platform/api/host/HostsOutputFilter.java

```diff
@@ -77,6 +78,13 @@ private Resource addLinks(Object original, Resource converted) {
             if (!add && StringUtils.isNotEmpty((String) converted.getFields().get(FIELD_DRIVER))) {
                 add = canAccessConfig();
             }
+            Object imported = DataAccessor.fromDataFieldOf(original).withKey(HostConstants.FIELD_IMPORTED).get();
+            if (imported != null && (Boolean)imported) {
```

Comment:
protip: `if (imported instanceof Boolean)`  that will ensure you don't get a classcastexception but also "null" is not an instanceof anything, so it checks for null too.